File: src/main/java/org/telran/lecture_02_big_o/practice/Factorial.java
package org.telran.lecture_02_big_o.practice;

// Задача: Вычислить факториал заданного числа.

// Алгоритм:
// Создать переменную для хранения результата и инициализировать ее единицей.
//     Умножить результат на все числа от 1 до заданного числа.
//     Вернуть результат.

public class Factorial {
    public static void main(String[] args) {
        System.out.println("factorial(0) - " + factorial(0)); // 1
        System.out.println("factorial(1) - " + factorial(1)); // 1
        System.out.println("factorial(2) - " + factorial(2)); // 2
        System.out.println("factorial(5) - " + factorial(5)); // 120
        System.out.println("factorial(10) - " + factorial(10)); // 3628800
    }

    public static int factorial(int n) {
        if (n == 0) return 1;
        int result = 1;
        for (int i = 2; i <= n; i++) {
            result *= i;
        }
        return result;
    }
    // 1*2*3*...*n
}

// Оценить сложность алгоритма
// O(n)

File: src/main/java/org/telran/lecture_02_big_o/practice/factorial.js
// Задача: Вычислить факториал заданного числа.

// Алгоритм:
// Создать переменную для хранения результата и инициализировать ее единицей.
//     Умножить результат на все числа от 1 до заданного числа.
//     Вернуть результат.

function factorial(number) {
    if(number == 0){
        return 1
    }
    let result = 1
    for (let i = 1; i <= number; i++) {
        result *= i   
    }
    return result
}
console.log(`factorial(0) - ${factorial(0)}`);
console.log(`factorial(1) - ${factorial(1)}`);
console.log(`factorial(2) - ${factorial(2)}`);
console.log(`factorial(4) - ${factorial(4)}`);
console.log(`factorial(10) - ${factorial(10)}`);

// Оценить сложность алгоритма O(n)
